homeassistant/components/switchbot_via_api/entity.py

```diff
@@ -0,0 +1,51 @@
+"""Base class for Switchbot via API entities."""
+import logging
+from typing import Any
+
+from switchbot_api import Device, Remote
+
+from homeassistant.helpers.entity import Entity
+
+from .const import API, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SwitchbotViaAPIEntity(Entity):
+    """Representation of a Switchbot via API entity."""
+
+    _is_remote = False
+    _switchbot_state: dict[str, Any] | None = None
+
+    def __init__(self, device: Device | Remote) -> None:
+        """Initialize the entity."""
+        super().__init__()
+        self._attr_unique_id = device.device_id
+        self._attr_name = device.device_name
+        self._is_remote = isinstance(device, Remote)
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, device.device_id)},
```

Comment:
Is the device_id globally unique?

Reply:
I think it is as it is a MAC address for physical devices, but let me confirm for remotes. I asked a question in Reddit:
https://www.reddit.com/r/TrySwitchBot/comments/16aeu44/about_deviceids/